cfn-resources/api-key/cmd/resource/resource.go

```diff
@@ -461,23 +461,24 @@ func areStringArraysEqualIgnoreOrder(arr1, arr2 []string) bool {
 	return true
 }
 
-func (model *Model) readAPIKeyDetails(apikey atlasSDK.ApiKeyUserDetails) Model {
+func (model *Model) readAPIKeyDetails(apikey admin.ApiKeyUserDetails) Model {
 	model.APIUserId = apikey.Id
 	model.Description = apikey.Desc
 	model.PublicKey = apikey.PublicKey
 	model.PrivateKey = apikey.PrivateKey
 	var roles []string
 	var projectAssignments []ProjectAssignment
 	var projectRolesMap = map[string][]string{}
-	for i := range apikey.Roles {
+	apiKeyRoles := apikey.GetRoles()
+	for i := range apiKeyRoles {
 		// org roles
-		if apikey.Roles[i].OrgId != nil && apikey.Roles[i].RoleName != nil {
-			roles = append(roles, *apikey.Roles[i].RoleName)
+		if apiKeyRoles[i].OrgId != nil && apiKeyRoles[i].RoleName != nil {
+			roles = append(roles, *apiKeyRoles[i].RoleName)
 		}
 		// project roles
-		if apikey.Roles[i].GroupId != nil {
-			if apikey.Roles[i].RoleName != nil {
-				projectRolesMap[*apikey.Roles[i].GroupId] = append(projectRolesMap[*apikey.Roles[i].GroupId], *apikey.Roles[i].RoleName)
+		if apiKeyRoles[i].GroupId != nil {
+			if apiKeyRoles[i].RoleName != nil {
+				projectRolesMap[*apiKeyRoles[i].GroupId] = append(projectRolesMap[*apiKeyRoles[i].GroupId], *apiKeyRoles[i].RoleName)
```

Comment:
can SDK Get functions like GetGroupId, GetRoleName, etc. be used?

Reply:
Prefer to leave this logic for handling the absence of specific properties in this case (vs using a potential empty string directly).